core/src/net/mod.rs

```diff
@@ -675,14 +688,23 @@ pub mod net_test_helpers {
                 Ok(pong) => {
                     debug!(self.ctx.log(), "Got msg {:?}", pong);
                     self.count += 1;
-                    if self.count < PING_COUNT {
-                        if self.eager {
+                    match self.count.cmp(&PING_COUNT) {
+                        Ordering::Less if self.eager => {
                             self.target
                                 .tell_serialised(PingMsg { i: pong.i + 1 }, self)
                                 .expect("serialise");
-                        } else {
+                        }
+                        Ordering::Less => {
                             self.target.tell(PingMsg { i: pong.i + 1 }, self);
                         }
+                        Ordering::Equal if self.promise.is_some() => {
+                            self.promise
+                                .take()
+                                .unwrap()
+                                .fulfil(())
+                                .expect("Failed to fulfil promise");
+                        }
+                        _ => {}
```

Comment:
I don't like "do nothing branches" without a comment in tests, in particular. Either this isn't supposed to do anything then it should be something to the effect of `_ => () // ignore` (a unit literal, *not* an empty block), **or** it is not supposed to ever be entered, then there should be `unreachable!("<reason>")` in there to explain why.